// helius-sdk/dist/esm/transactions/types.js
const SENDER_ENDPOINTS = {
    Default: "https://sender.helius-rpc.com",
    US_SLC: "http://slc-sender.helius-rpc.com",
    US_EAST: "http://ewr-sender.helius-rpc.com",
    EU_WEST: "http://lon-sender.helius-rpc.com",
    EU_CENTRAL: "http://fra-sender.helius-rpc.com",
    EU_NORTH: "http://ams-sender.helius-rpc.com",
    AP_SINGAPORE: "http://sg-sender.helius-rpc.com",
    AP_TOKYO: "http://tyo-sender.helius-rpc.com",
};
// `/fast` endpoint used for sending transactions
const senderFastUrl = (region) => `${SENDER_ENDPOINTS[region]}/fast`;
// `/ping` endpoint used for connection warming
const senderPingUrl = (region) => `${SENDER_ENDPOINTS[region]}/ping`;
const SENDER_TIP_ACCOUNTS = [
    "4ACfpUFoaSD9bfPdeu6DBt89gB6ENTeHBXCAi87NhDEE",
    "D2L6yPZ2FmmmTKPgzaMKdhu6EWZcTpLy1Vhx8uvZe7NZ",
    "9bnz4RShgq1hAnLnZbP8kbgBg1kEmcJBYQq3gQbmnSta",
    "5VY91ws6B2hMmBFRsXkoAAdsPHBJwRfBht4DXox3xkwn",
    "2nyhqdwKcJZR2vcqCyrYsaPVdAnFoJjiksCXJ7hfEYgD",
    "2q5pghRs6arqVjRvT5gfgWfWcHWmw1ZuCzphgd5KfWGJ",
    "wyvPkWjVZz1M8fHQnMMCDTQDbkManefNNhweYk5WkcF",
    "3KCKozbAaF75qEU33jtzozcJ29yJuaLJTy2jFdzUY8bT",
    "4vieeGHPYPG2MmyPRcYjdiDmmhN3ww7hsFNap8pVN3Ey",
    "4TQLFNWK8AovT1gFvda5jfw2oJeRMKEmw7aH6MGBJ3or",
];
const MIN_TIP_LAMPORTS_DUAL = 1000000n; // 0.001 SOL
const MIN_TIP_LAMPORTS_SWQOS = 500000n; // 0.0005 SOL
const SFDP_REJECTS_URL = "https://helius-docs.s3.us-east-2.amazonaws.com/sfdp_rejects.json";

export { MIN_TIP_LAMPORTS_DUAL, MIN_TIP_LAMPORTS_SWQOS, SENDER_ENDPOINTS, SENDER_TIP_ACCOUNTS, SFDP_REJECTS_URL, senderFastUrl, senderPingUrl };
