// helius-sdk/dist/cjs/rpc/createHelius.eager.cjs
'use strict';

var kit = require('@solana/kit');
var wrapAutoSend = require('./wrapAutoSend.cjs');
var caller = require('./caller.cjs');
var getAsset = require('./methods/getAsset.cjs');
var getAssetBatch = require('./methods/getAssetBatch.cjs');
var getAssetProof = require('./methods/getAssetProof.cjs');
var getAssetProofBatch = require('./methods/getAssetProofBatch.cjs');
var getAssetsByAuthority = require('./methods/getAssetsByAuthority.cjs');
var getAssetsByCreator = require('./methods/getAssetsByCreator.cjs');
var getAssetsByGroup = require('./methods/getAssetsByGroup.cjs');
var getPriorityFeeEstimate = require('./methods/getPriorityFeeEstimate.cjs');
var client_eager$2 = require('../webhooks/client.eager.cjs');
var getAssetsByOwner = require('./methods/getAssetsByOwner.cjs');
var getNftEditions = require('./methods/getNftEditions.cjs');
var getSignaturesForAsset = require('./methods/getSignaturesForAsset.cjs');
var getTokenAccounts = require('./methods/getTokenAccounts.cjs');
var searchAssets = require('./methods/searchAssets.cjs');
var client_eager$1 = require('../enhanced/client.eager.cjs');
var client_eager = require('../transactions/client.eager.cjs');
var getProgramAccountsV2 = require('./methods/getProgramAccountsV2.cjs');
var getAllProgramAccounts = require('./methods/getAllProgramAccounts.cjs');
var getTokenAccountsByOwnerV2 = require('./methods/getTokenAccountsByOwnerV2.cjs');
var getAllTokenAccountsByOwner = require('./methods/getAllTokenAccountsByOwner.cjs');
var getTransactionsForAddress = require('./methods/getTransactionsForAddress.cjs');

const createHeliusEager = ({ apiKey, network = "mainnet", rebateAddress, }) => {
    const rebateParam = rebateAddress ? `&rebate-address=${rebateAddress}` : "";
    const url = `https://${network}.helius-rpc.com/?api-key=${apiKey}${rebateParam}`;
    const solanaApi = kit.createSolanaRpcApi(kit.DEFAULT_RPC_CONFIG);
    const transport = kit.createDefaultRpcTransport({ url });
    let baseRpc = kit.createRpc({ api: solanaApi, transport });
    // Cast to any because I cba to go down this type rabbit hole
    const raw = wrapAutoSend.wrapAutoSend(baseRpc);
    const call = caller.makeRpcCaller(transport);
    return {
        raw,
        // DAS
        getAsset: getAsset.makeGetAsset(call),
        getAssetBatch: getAssetBatch.makeGetAssetBatch(call),
        getAssetProof: getAssetProof.makeGetAssetProof(call),
        getAssetProofBatch: getAssetProofBatch.makeGetAssetProofBatch(call),
        getAssetsByAuthority: getAssetsByAuthority.makeGetAssetsByAuthority(call),
        getAssetsByCreator: getAssetsByCreator.makeGetAssetsByCreator(call),
        getAssetsByGroup: getAssetsByGroup.makeGetAssetsByGroup(call),
        getAssetsByOwner: getAssetsByOwner.makeGetAssetsByOwner(call),
        getNftEditions: getNftEditions.makeGetNftEditions(call),
        getTokenAccounts: getTokenAccounts.makeGetTokenAccounts(call),
        getSignaturesForAsset: getSignaturesForAsset.makeGetSignaturesForAsset(call),
        searchAssets: searchAssets.makeSearchAssets(call),
        // Priority Fee API
        getPriorityFeeEstimate: getPriorityFeeEstimate.makeGetPriorityFeeEstimate(call),
        // V2 methods
        getProgramAccountsV2: getProgramAccountsV2.makeGetProgramAccountsV2(call),
        getAllProgramAccounts: getAllProgramAccounts.makeGetAllProgramAccounts(call),
        getTokenAccountsByOwnerV2: getTokenAccountsByOwnerV2.makeGetTokenAccountsByOwnerV2(call),
        getAllTokenAccountsByOwner: getAllTokenAccountsByOwner.makeGetAllTokenAccountsByOwner(call),
        getTransactionsForAddress: getTransactionsForAddress.makeGetTransactionsForAddress(call),
        // Webhooks
        webhooks: client_eager$2.makeWebhookClientEager(apiKey),
        // Enhanced Transactions
        enhanced: client_eager$1.makeEnhancedTxClientEager(apiKey, network),
        // Transaction helpers
        tx: client_eager.makeTxHelpersEager(baseRpc),
    };
};

exports.createHeliusEager = createHeliusEager;
