// helius-sdk/dist/esm/rpc/createHelius.eager.js
import { createSolanaRpcApi, DEFAULT_RPC_CONFIG, createDefaultRpcTransport, createRpc } from '@solana/kit';
import { wrapAutoSend } from './wrapAutoSend.js';
import { makeRpcCaller } from './caller.js';
import { makeGetAsset } from './methods/getAsset.js';
import { makeGetAssetBatch } from './methods/getAssetBatch.js';
import { makeGetAssetProof } from './methods/getAssetProof.js';
import { makeGetAssetProofBatch } from './methods/getAssetProofBatch.js';
import { makeGetAssetsByAuthority } from './methods/getAssetsByAuthority.js';
import { makeGetAssetsByCreator } from './methods/getAssetsByCreator.js';
import { makeGetAssetsByGroup } from './methods/getAssetsByGroup.js';
import { makeGetPriorityFeeEstimate } from './methods/getPriorityFeeEstimate.js';
import { makeWebhookClientEager } from '../webhooks/client.eager.js';
import { makeGetAssetsByOwner } from './methods/getAssetsByOwner.js';
import { makeGetNftEditions } from './methods/getNftEditions.js';
import { makeGetSignaturesForAsset } from './methods/getSignaturesForAsset.js';
import { makeGetTokenAccounts } from './methods/getTokenAccounts.js';
import { makeSearchAssets } from './methods/searchAssets.js';
import { makeEnhancedTxClientEager } from '../enhanced/client.eager.js';
import { makeTxHelpersEager } from '../transactions/client.eager.js';
import { makeGetProgramAccountsV2 } from './methods/getProgramAccountsV2.js';
import { makeGetAllProgramAccounts } from './methods/getAllProgramAccounts.js';
import { makeGetTokenAccountsByOwnerV2 } from './methods/getTokenAccountsByOwnerV2.js';
import { makeGetAllTokenAccountsByOwner } from './methods/getAllTokenAccountsByOwner.js';
import { makeGetTransactionsForAddress } from './methods/getTransactionsForAddress.js';

const createHeliusEager = ({ apiKey, network = "mainnet", rebateAddress, }) => {
    const rebateParam = rebateAddress ? `&rebate-address=${rebateAddress}` : "";
    const url = `https://${network}.helius-rpc.com/?api-key=${apiKey}${rebateParam}`;
    const solanaApi = createSolanaRpcApi(DEFAULT_RPC_CONFIG);
    const transport = createDefaultRpcTransport({ url });
    let baseRpc = createRpc({ api: solanaApi, transport });
    // Cast to any because I cba to go down this type rabbit hole
    const raw = wrapAutoSend(baseRpc);
    const call = makeRpcCaller(transport);
    return {
        raw,
        // DAS
        getAsset: makeGetAsset(call),
        getAssetBatch: makeGetAssetBatch(call),
        getAssetProof: makeGetAssetProof(call),
        getAssetProofBatch: makeGetAssetProofBatch(call),
        getAssetsByAuthority: makeGetAssetsByAuthority(call),
        getAssetsByCreator: makeGetAssetsByCreator(call),
        getAssetsByGroup: makeGetAssetsByGroup(call),
        getAssetsByOwner: makeGetAssetsByOwner(call),
        getNftEditions: makeGetNftEditions(call),
        getTokenAccounts: makeGetTokenAccounts(call),
        getSignaturesForAsset: makeGetSignaturesForAsset(call),
        searchAssets: makeSearchAssets(call),
        // Priority Fee API
        getPriorityFeeEstimate: makeGetPriorityFeeEstimate(call),
        // V2 methods
        getProgramAccountsV2: makeGetProgramAccountsV2(call),
        getAllProgramAccounts: makeGetAllProgramAccounts(call),
        getTokenAccountsByOwnerV2: makeGetTokenAccountsByOwnerV2(call),
        getAllTokenAccountsByOwner: makeGetAllTokenAccountsByOwner(call),
        getTransactionsForAddress: makeGetTransactionsForAddress(call),
        // Webhooks
        webhooks: makeWebhookClientEager(apiKey),
        // Enhanced Transactions
        enhanced: makeEnhancedTxClientEager(apiKey, network),
        // Transaction helpers
        tx: makeTxHelpersEager(baseRpc),
    };
};

export { createHeliusEager };
